code/modules/mob/living/maneuvers/maneuver_leap.dm

```diff
@@ -12,13 +12,19 @@
 		strength = max(2, strength * user.get_jump_distance())
 		if(reflexively)
 			strength *= reflexive_modifier
+		var/start_layer = user.layer
+		user.layer = VEHICLE_LOAD_LAYER
 		animate(user, pixel_z = 16, time = 3, easing = SINE_EASING | EASE_IN)
 		animate(pixel_z = user.default_pixel_z, time = 3, easing = SINE_EASING | EASE_OUT)
 		user.throw_at(get_turf(target), strength, 1, user, FALSE, CALLBACK(src, /decl/maneuver/leap/proc/end_leap, user, target, old_pass_flags))
+		addtimer(CALLBACK(src, .proc/reset_layer, start_layer, user), 4)
 
 /decl/maneuver/leap/proc/end_leap(var/mob/living/user, var/atom/target, var/pass_flag)
 	user.pass_flags = pass_flag
 
+/decl/maneuver/leap/proc/reset_layer(start_layer, mob/living/user)
+	user.layer = start_layer
```

Comment:
This is closer to being reasonable, but you should make the mob recompute its layer, not use the cached layer.